Game/Src/Powers/Powers.gd

```diff
@@ -0,0 +1,38 @@
+extends Node
+
+onready var lightning = $LightningPower
+onready var potion = $PotionPower
+onready var slippery = $SlipperyHandsPower
+
+var rng
+
+signal glory_increment(value)
+signal health_increase(value)
+
+func _ready() -> void:
+	randomize()
+	rng = RandomNumberGenerator.new()
+
+func call_pwr(slots,pwrList,toggle,otherArgs: Dictionary = {'stats':null,'enemy':null}):
```

Comment:
It is not entirely clear but I think this is exactly the case that I mentioned with the apply_power function.
You could just delete this function and call power.apply_function instead in TestEnemymvt

Reply:
This is a new generic function that is called whenever a power is invoked and handles which power is to be used. otherArgs is a dictionnary that contains any additional information that may be needed.
It might be time I comment what I do..